Parquet/src/main/java/io/deephaven/parquet/ParquetFileWriter.java

```diff
@@ -68,19 +73,19 @@ public RowGroupWriter addRowGroup(long size) {
     }
 
     public void close() throws IOException {
+        final OutputStream os = Channels.newOutputStream(writeChannel);
+        os.write(ParquetFileReader.MAGIC);
         serializeOffsetIndexes(offsetIndexes, blocks, writeChannel);
         ParquetMetadata footer = new ParquetMetadata(new FileMetaData(type, extraMetaData, Version.FULL_VERSION), blocks);
-        serializeFooter(footer, writeChannel);
+        serializeFooter(footer, os);
         writeChannel.close();
```

Comment:
Switch to a `try-finally` and close `os` instead of `writeChannel`. See JavaDoc for `Channels.newOutputStream`.